src/modules/security.js

```diff
@@ -512,11 +509,82 @@
       errorMessageKey: 'requiredFields'
     });
 
+    /*
+     * Google reCaptcha 2
+     */
+    $.formUtils.addValidator({
+        name: 'recaptcha',
+        validatorFunction: function (val, $el)
+        {
+            return grecaptcha.getResponse($el.data('validation-recaptcha-widget-id'));
+        },
+        errorMessage: '',
+        errorMessageKey: 'badreCaptcha'
+    });
 
-
-  $.fn.displayPasswordStrength = function(conf) {
+    $.fn.displayPasswordStrength = function(conf) {
         new $.formUtils.validators.validate_strength.strengthDisplay(this, conf);
         return this;
     };
 
+    var setupGooglereCaptcha = function (evt, $forms, config)
+    {
+        if( typeof grecaptcha !== typeof undefined ){
+            throw new Error('reCaptcha API can not be loaded by hand, delete reCaptcha API snippet.');
+        }
+
```

Comment:
Is this really necessary? This might cause unexpected crashes since the method is attached to `modulesLoaded` and `formValidationSetup`. These two events may be dispatched several times during a page load.

If it feels necessary you will need to use a global variable to check whether or not the grecaptcha was inserted by the jquery-form-validator. Something like:

```js
if (typeof grecaptcha !== typeof undefined && !$.formUtils.hasLoadedGrecaptcha) {
   throw Error...
} else if(!$.formUtils.hasLoadedGrecaptcha) {
   $.formUtils.hasLoadedGrecaptcha = true;
   // Insert script...
}
```